Gen 4 TID/SID guide (#308)
zaksabeast/PokemonRNGGuides: guides/Gen 4/Diamond, Pearl, and Platinum/TID SID.mdx

---
title: "Diamond, Pearl, and Platinum TID/SID RNG"
navDrawerTitle: "TID/SID RNG"

> description: "How to get a desired TID/SID combo"

Reviewer: Let's change this back to `Learn how to obtain your desired Trainer ID (TID) and Secret ID (SID) combo in Diamond, Pearl, and Platinum.`.  That's SEO optimized, so people using Google and other search engines will have an easier time finding this page.

slug: "emulator-dppt-tid-sid"
category: "Diamond, Pearl, and Platinum"

> isRoughDraft: false

Reviewer: > I couldn't figure out how to get this guide to move from rough draft area as I used the draft document as a base to add my guide on top basically.

This is almost there!  The last thing to do is run `bun run build:guides`, which will update `src/__generated__/guides.ts` to have `isRoughDraft: false`.

tag: "emu"
---

## Tools

> - [PokeFinder](https://github.com/Admiral-Fish/PokeFinder/releases)

Reviewer: We found out several people aren't sure what PokeFinder is, and there are often questions about how to set it up after downloading it.

We created a [`/pokefinder`](https://www.pokemonrng.com/pokefinder/) page to clear things up for users before sending them to GitHub.

Let's update this to use the new link:
```suggestion
- [PokeFinder](/pokefinder)
```

- [Desmume](/desmume-setup)
- [RunAsDate](https://www.nirsoft.net/utils/run_as_date.html)

## Step 1: Choosing your ID to hit

1. Go to the Gen 4 tab and select "ID's"

2. Under Filters, select TID and enter any 5-digit number you want

>    ![pk-finder-id-boxes](/images/Diamond-Pearl-Platinum/TID/pkfinder-ID-search.png)

Reviewer: What are thoughts on moving this below these four instructions since it does a good job demonstrating what the user sees across multiple of these?

Author: never mind, actually did it and i don't think it looks nice. your suggestion looks better. i can't find a better spot for the note. it feels miss-able now but i can't think of a solution.

> 1. Go to the Gen 4 tab and select "ID's"
> 
> 2. Under Filters, select TID and enter any 5-digit number you want
> 
>    ![pk-finder-id-boxes](/images/Diamond-Pearl-Platinum/TID/pkfinder-ID-search.png)
> 
> 3. Choose a TID and SID combo that you like
> 
> 4. Right click on chosen option and select "Generates time for seed"

Reviewer: Lets make a few changes here:
- Remove the spaces in between each line
- Add full stops to the end of each
- A tiny word change on line 4

```suggestion
1. Go to the Gen 4 tab and select "ID's".
2. Under Filters, select TID and enter any 5-digit number you want.
   ![pk-finder-id-boxes](/images/Diamond-Pearl-Platinum/TID/pkfinder-ID-search.png)
3. Choose a TID and SID combo that you like.
4. Right click on the chosen option and select "Generate time for seed".
```

```

> Make sure your delay is at least 5000 frames, higher just gives you more time to hit your frame.

Reviewer: Let's use the word "advances" here since we're trying to move the community that direction:
```suggestion
Make sure your delay is at least 5000 advances, higher just gives you more time to hit your advance.
```

```

> ## Step 3: Setup RunAsDate

Reviewer: It looks like we're missing Step 2.  Let's make this Step 2 and the next one Step 3.

1. Launch RunAsDate.
2. Configure RunAsDate to match the image below.

![Setup](/images/Diamond-Pearl-Platinum/Initial-Seed/Setup.png)

```
You'll never have to change that after. This is the universal RunAsDate configuration for RNG (Gen 3, 4 or 5).
```

3. Select the program you want to fake the date/time with the `Browse...` button.
4. Set the date and time to hit your seed as shown in PokeFinder.
   - Right click on the chosen seed and choose "Generate times for seed".

> 5. Hit `Run` to launch Desmume.

Reviewer: Let's update this section to follow the instructions at `/dppt-initial-seed` since it has the same content.

This way if we update how initial seeds are hit we only need to update one guide.

## Step 4: Hit delay

Now that you have set up run as date and started your game, you will want to get through the opening sequence as fast as possible.

```
Note: When naming your character, you will be changing your delay between even and odds if you use Capital letters or add symbols to your name. Keep this in mind, if you need to redo this step.
```

> Once you have gotten to this screen below, press "N" and make a save state here. then slowly get to you target delay and hold A as you press CTRL + P

Reviewer: Since we use single ticks for button presses like `A + Start` and double quotes for clicks like "Generate", let's format button presses with single ticks and use a full stop:
```suggestion
Once you have gotten to this screen below, press `N` and make a save state here. then slowly get to you target delay and hold `A` as you press `CTRL + P`.
```

Platinum:
![pt-last-screen](/images/Diamond-Pearl-Platinum/TID/pt-last-screen-tid.png)
Diamond and Pearl:

> ![dp-last-screen](/images/Diamond-Pearl-Platinum/TID/DP-last-screen-tid.png)

Reviewer: These screenshots are a nice touch.

Not a requirement, but a possible change: it might look a little cleaner to show this in a table.

The markdown would look like this:
```md
| Platinum                                                                     | Diamond and Pearl                                                            |
| ---------------------------------------------------------------------------- | ---------------------------------------------------------------------------- |
| ![pt-last-screen](/images/Diamond-Pearl-Platinum/TID/pt-last-screen-tid.png) | ![dp-last-screen](/images/Diamond-Pearl-Platinum/TID/DP-last-screen-tid.png) |
```

And here's the result:
| Desktop | Mobile | Mobile with expanded image |
| - | - | - |
| ![localhost_5173_emulator-dppt-tid-sid(M1 Mac)](https://github.com/user-attachments/assets/d973c1bd-cff5-4510-a862-c402338be40b) | ![localhost_5173_emulator-dppt-tid-sid(Pixel 7)](https://github.com/user-attachments/assets/c31827ff-2055-45f5-8e9a-5cc7fd5b2771) | ![localhost_5173_emulator-dppt-tid-sid(Pixel 7) (1)](https://github.com/user-attachments/assets/3ed59bca-3d39-4e34-9ba0-95c82288554d) |

Thoughts?

You will most likely have gotten the wrong TID. Enter the TID you did get in seed finder like below:

![hit-tid](/images/Diamond-Pearl-Platinum/TID/hit-tid-pkfinder-ex.png)

In this example, I hit 46 early. Go back to your save state and then hit your delay but adjust for the delay you experienced. 

Congrats! You should now have your target ID.

- If you experience issues with the delay swapping from even and odd, you can go up or down a year on the date pokefinder gives you to swap as well.
- The PID option is a useful filter option, if you want to tie this RNG into RNG'ing your starter as you can get a good stat spread to be shiny that has a good delay.
- or you could use a cute charm PID to go do a cute charm RNG next!

> [Cute Charm RNG guide](/emulator-dppt-cute-charm)

Reviewer: Great job doing this - giving people a sense of what to do next is 💯

Author: Thank you! I figured it would help people who might be like me sometimes who do something and then aren't sure what to do next lol
